core/src/layers/minitrace.rs

```diff
@@ -0,0 +1,405 @@
+    }
+}
+
+#[derive(Debug)]
+pub struct MinitraceAccessor<A> {
+    inner: A,
+}
+
+#[async_trait]
+impl<A: Accessor> LayeredAccessor for MinitraceAccessor<A> {
+    type Inner = A;
+    type Reader = MinitraceWrapper<A::Reader>;
+    type BlockingReader = MinitraceWrapper<A::BlockingReader>;
+    type Writer = MinitraceWrapper<A::Writer>;
+    type BlockingWriter = MinitraceWrapper<A::BlockingWriter>;
+    type Pager = MinitraceWrapper<A::Pager>;
+    type BlockingPager = MinitraceWrapper<A::BlockingPager>;
+
+    fn inner(&self) -> &Self::Inner {
+        &self.inner
+    }
+
+    #[trace("metadata")]
+    fn metadata(&self) -> AccessorInfo {
+        self.inner.info()
+    }
+
+    #[trace("create", enter_on_poll = true)]
+    async fn create(&self, path: &str, args: OpCreate) -> Result<RpCreate> {
+        self.inner.create(path, args).await
+    }
+
+    async fn read(&self, path: &str, args: OpRead) -> Result<(RpRead, Self::Reader)> {
+        self.inner
+            .read(path, args)
+            .map(|v| {
+                v.map(|(rp, r)| {
+                    (
+                        rp,
+                        MinitraceWrapper::new(Span::enter_with_local_parent("ReadOperation"), r),
+                    )
+                })
+            })
+            .enter_on_poll("read")
+            .await
+    }
+
+    async fn write(&self, path: &str, args: OpWrite) -> Result<(RpWrite, Self::Writer)> {
+        self.inner
+            .write(path, args)
+            .map(|v| {
+                v.map(|(rp, r)| {
+                    (
+                        rp,
+                        MinitraceWrapper::new(Span::enter_with_local_parent("WriteOperation"), r),
+                    )
+                })
+            })
+            .enter_on_poll("write")
+            .await
+    }
+
+    #[trace("stat", enter_on_poll = true)]
+    async fn stat(&self, path: &str, args: OpStat) -> Result<RpStat> {
+        self.inner.stat(path, args).await
+    }
+
+    #[trace("delete", enter_on_poll = true)]
+    async fn delete(&self, path: &str, args: OpDelete) -> Result<RpDelete> {
+        self.inner.delete(path, args).await
+    }
+
+    async fn list(&self, path: &str, args: OpList) -> Result<(RpList, Self::Pager)> {
+        self.inner
+            .list(path, args)
+            .map(|v| {
+                v.map(|(rp, s)| {
+                    (
+                        rp,
+                        MinitraceWrapper::new(Span::enter_with_local_parent("PageOperation"), s),
+                    )
+                })
+            })
+            .enter_on_poll("list")
+            .await
+    }
+
+    async fn scan(&self, path: &str, args: OpScan) -> Result<(RpScan, Self::Pager)> {
+        self.inner
+            .scan(path, args)
+            .map(|v| {
+                v.map(|(rp, s)| {
+                    (
+                        rp,
+                        MinitraceWrapper::new(Span::enter_with_local_parent("PageOperation"), s),
```

Comment:
![image](https://user-images.githubusercontent.com/5351546/231641491-35f40fde-2427-47c4-9244-e19bcdc34977.png)

This seems add a new span. Can we reuse the same span in `scan`?

Reply:
ok, remove enter_on_poll